llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -526,6 +526,10 @@ bool llvm::isValidAssumeForContext(const Instruction *Inv,
     return AllowEphemerals || !isEphemeralValueOf(Inv, CxtI);
   }
 
+  // Inv and CxtI are in different functions.
+  if (Inv->getFunction() != CxtI->getFunction())
```

Comment:
Which caller has a context from a different function?

Reply:
The alignment-from-assumptions pass iterates over all users of alignment pointers of calls with an alignment assumption and calls `isValidAssumeForContext` with the the call instruction and the user of the ptr -- which can be a global, in which case users can be in different functions. (This actually happened. It reminds me that I should add a test case, because the tests didn't catch this -- only a stage2 LLVM compilation did. Will add a test case tomorrow.)

Reply:
We should check for this inside alignment-from-assumptions, not here.